src/logger.ts

```diff
@@ -0,0 +1,17 @@
+import { Logger, makeConsoleLogFactory } from "@graphile/logger";
+
+export { LogLevel as MigrateLogLevel } from "@graphile/logger";
+export { Logger as MigrateLogger } from "@graphile/logger";
+export { LogFunctionFactory as MigrateLogFactory } from "@graphile/logger";
+export interface MigrateLogMeta {
+  error?: Error;
+}
```

Comment:
We're actually not using this anywhere, so let's remove it.

```suggestion
```

Reply:
Let's say that right now `@graphile/migrate` only logs `error` as part of the LogMeta, but in the future, it starts logging something else, e.g some random `context` property. 

On one side, if this interface is kept - it needs explicit updates whenever LogMeta properties change somewhere in the code, but it's not tied to the logger calls in any way, so it's easy to forget to update it. From this perspective, it makes sense to drop it.

On the other side, if such an interface is provided - it will let consumers of `@graphile/migrate` know with a certainty that LogMeta contains a new `context` property and use it (or not use it). It also provides info that `error` is actually can be part of the meta (and not something like `exception`).

With this in mind, do you think it's still better to drop the interface?

Reply:
Or to put it into integration perspective, the consumer would have to define the interface (e.g. depending on release notes or @graphile/migrate source code inspection) :)

```ts
import { Logger} from 'some-custom-logger-class';
import {
  LogFunctionFactory,
  Logger as MigrateLogger,
  LogLevel,
} from 'graphile-migrate';

//....

interface MigrateLogMeta {
  error?: Error;
}

//....

const logger = new Logger('Migrations');
const logFactory: LogFunctionFactory<Record<string, unknown>> = () => {
  return (
    level: LogLevel,
    message: string,
    meta?: MigrateLogMeta,
  ): void => {
    switch (level) {
      case LogLevel.ERROR:
        meta?.error
          ? logger.error(meta?.error, message)
          : logger.error(message);
        break;
      case LogLevel.WARNING:
        logger.warn(message);
        break;
      case LogLevel.INFO:
      default:
        logger.log(message);
    }
  };
};
const migrateLogger = new MigrateLogger(logFactory); // passed to settings
```

Reply:
Actually `Logger` accepts a log scope; so we should do something like we do in Graphile Worker where the scope is automatically applied:

https://github.com/graphile/worker/blob/6a6634d82e64b87e7e33b68819debde383344ee2/src/logger.ts#L12-L24

Then remove the `@graphile/logger` re-exports and only export the scoped versions.

Reply:
>Then remove the @graphile/logger re-exports and only export the scoped versions.

I would postpone this for some different PR since right there are no concrete scoped variables. (e.g. if all log message texts are refactored as some point)

And scope vs meta topic we discussed in discord :)